FunctionalExtensions

RomanLevashev opened the pull request.

RomanLevashev committed:
FunctionalExtensions

YuriUfimtsev commented on Utils/ListFunctions/FunctionalExtensions.cs:
```diff
@@ -0,0 +1,71 @@
+        /// <returns>The final accumulator value.</returns>
+        public static TAccumulate Fold<T, TAccumulate>(this IEnumerable<T> source, TAccumulate seed, Func<TAccumulate, T, TAccumulate> func)
+        {
+            TAccumulate accumulator = seed;
```
seed, оригинально :)

YuriUfimtsev commented on Utils/ListFunctions/FunctionalExtensions.cs:
```diff
@@ -0,0 +1,71 @@
+﻿namespace ListFunctions
+{
```
Можно использовать file-scoped namespace

YuriUfimtsev commented on Utils/ListFunctionsTest/FunctionalExtensionsTest.cs:
```diff
@@ -0,0 +1,50 @@
+﻿namespace ListFunctionsTest
+{
```
И здесь

YuriUfimtsev commented on Utils/ListFunctionsTest/FunctionalExtensionsTest.cs:
```diff
@@ -0,0 +1,50 @@
+﻿namespace ListFunctionsTest
+{
+    using ListFunctions;
+
```
Раз функции с генерик-типами, стоит здесь проверить ещё на списках со строками, например.
И на пустых списках хорошо бы проверить все методы

RomanLevashev committed:
Merge branch 'main' into Utils

RomanLevashev committed:
- Add license headers to all source files
- Include stylecop.json configuration
- Add tests for string lists and empty lists scenarios

yurii-litvinov commented on Utils/.editorconfig:
```diff
@@ -0,0 +1,7 @@
+﻿[*.cs]
+
+# SA1633: File should have header
+dotnet_diagnostic.SA1633.severity = none
+
+# SA1010: Opening square brackets should not be preceded by a space.
+dotnet_diagnostic.SA1010.severity = none
```
Это уже не надо выключать, шапки в файлах же появились

yurii-litvinov commented on Utils/ListFunctions/ListFunctions.csproj:
```diff
@@ -0,0 +1,20 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
```
:(

yurii-litvinov commented on Utils/ListFunctions/ListFunctions.csproj:
```diff
@@ -0,0 +1,20 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
```
Используйте 1.2-beta

yurii-litvinov commented on Utils/ListFunctionsTest/FunctionalExtensionsTest.cs:
```diff
@@ -0,0 +1,132 @@
+    [TestMethod]
+    public void MapWithIntTest()
+    {
+        Func<int, int> plus = (x) => x + 2;
```
```suggestion
        Func<int, int> plus = x => x + 2;
```

yurii-litvinov reviewed (approved):
В целом всё ок, зачтена
